apps/settings/test/unit/panels/screen_lock_passcode/screen_lock_passcode_test.js

```diff
@@ -122,10 +146,16 @@ suite('ScreenLockPasscode > ', function() {
     });
 
     suite('passcode is the same', function() {
-      setup(function() {
-        screenLockPasscode._settings.passcode = '0000';
-        screenLockPasscode._passcodeBuffer = '0000';
-        screenLockPasscode._checkPasscode();
+      setup(function(done) {
+        // set passcode to 0000
+        var promise = PasscodeHelper.set('0000');
```

Comment:
Per the comments above, we can make `PasscodeHelper.check` return a promise resolved with true.